x-pack/plugins/security_solution/public/flyout/shared/utils/fetch_data.ts

```diff
@@ -15,10 +15,14 @@ export const createFetchData = async <TResponse, T = {}>(
   searchService: ISearchStart,
   req: IEsSearchRequest
 ): Promise<TResponse> => {
+  let rawResponse: TResponse;
   return new Promise((resolve, reject) => {
     searchService.search<IEsSearchRequest, IKibanaSearchResponse<TResponse>>(req).subscribe({
       next: (response) => {
-        resolve(response.rawResponse);
+        rawResponse = response.rawResponse;
+      },
+      complete: () => {
```

Comment:
It's sad that `response` is not passed to the `complete` callback...

Reply:
why?

Reply:
you can always do your own wrapper :D

Reply:
@PhilippeOberti If `response` was passed in the `complete` callback, we wouldn't have to keep a local reference of `rawResponse`.

Reply:
> @PhilippeOberti If `response` was passed in the `complete` callback, we wouldn't have to keep a local reference of `rawResponse`. 

 It then we'd have to keep a reference of the whole response object, which would be almost the same no?
The reason I thought I'd need to need to keep a reference, was because I need to return the response only in the complete callback and not the next callback. Returning in the next callback causes the issue when streaming, and on the complete callback we're not getting the response.
I coils have missed something though 😂

Reply:
Well, I guess it's a "problem" of observables being very generic and they don't care about your business logic, e.g. response values. If `complete` would be of the following type `(lastResponse: Response) => void`, we wouldn't need to keep track of the object ourselves.

Reply:
yeah I see what you mean now. I feel like the problem isn't really on the observables side here, but rather on React Query which doesn't really support them...If useQuery supported observables, we could carry the stream all the way to the frontend (like any Angular app using rxjs does) and we wouldn't care about `next` or `complete`... But yeah, for now it's a bit annoying...